### PR TITLE
Remove the dependency on python, add .cmd support for Windows

### PR DESCRIPTION
**Pull request by petemounce**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,5 +2,5 @@ local_run: vendorize
 	bk local run .buildkite/local-pipeline.yml
 
 vendorize:
-	mkdir .buildkite/plugins/metahook -p
-	cp hooks python plugin.yml .buildkite/plugins/metahook/ -rf
+	mkdir -p .buildkite/plugins/metahook
+	cp -rf hooks python plugin.yml .buildkite/plugins/metahook/
```
Comment (ca-johnson) [inline Makefile:6]: Dunno how those flags ended up there

File: hooks/environment [hunk 2]
```diff
@@ -1,2 +1,35 @@
...
+    "post-artifact"
+    "pre-exit"
+)
+declare -a extensions=(
+    ""
+    ".bat"
+    ".cmd"
+)
+echo "whereami $(pwd)"
+pwd
```
Comment (ca-johnson) [inline hooks/environment:21]:
```suggestion
```
Seems like this is for debugging?

File: hooks/environment [hunk 3]
```diff
@@ -1,2 +1,35 @@
...
+hooks_dir="$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"
+echo "hooks_dir: ${hooks_dir}"
+env | sort | grep METAHOOK
```
Comment (ca-johnson) [inline hooks/environment:24]:
```suggestion
```
Also seems like a debug line - its not the users business that we look this stuff up from env vars
(also its in buildkite env tab already)

File: contributing.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Contributing
```
Comment (ca-johnson) [inline contributing.md:1]: Nice! 👍

File: Makefile (hunk 1 above)
Comment (petemounce) [reply Makefile:6]: Shrug

File: hooks/environment [hunk 5]
```diff
@@ -1,2 +1,34 @@
...
+env | sort | grep "METAHOOK" | uniq >"${vars}"
+for hook_name in "${hooks[@]}"; do
+  for ext in "${extensions[@]}"; do
+    hook_file="${hooks_dir}/${hook_name}${ext}"
+    rm -f "${hook_file}"
+    upperd="$(echo "${hook_name}${ext}" | tr "[:lower:]" "[:upper:]")"
+    bkv="BUILDKITE_PLUGIN_METAHOOK_${upperd//-/_}"
+    echo "bkv: ${bkv}"
+    if grep "${bkv}" <"${vars}"; then
+      echo "${!bkv}" >"${hook_file}"
```
Comment (Helcaraxan) [inline hooks/environment:31]: You might want to add a note here as indirect variable expansion is a useful but rarely used feature that only a handful of people will know without looking it up.

File: hooks/environment [hunk 6]
```diff
@@ -1,2 +1,34 @@
-#!/bin/bash
-python "${BASH_SOURCE%/*}/../python/setup-hooks.py"
+#!/usr/bin/env bash
```
Comment (Helcaraxan) [inline hooks/environment:1]: Maybe add one or two empty lines to aerate the script.

Comment (jared-improbable) [reply hooks/environment:1]: "aerate", I love it :)

**Pull request merged**